src/pcap/cmd/pcap-api/main.go

```diff
@@ -33,42 +32,62 @@ func main() {
 	}
 
 	if err != nil {
-		log.Fatal("unable to initialize", zap.Error(err))
+		log.Panic("Unable to initialize", zap.Error(err))
 	}
 
 	err = config.validate()
 	if err != nil {
-		log.Fatal("unable to validate config", zap.Error(err))
+		log.Panic("Failed to validate config", zap.Error(err))
 	}
 
-	cmd.SetLogLevel(log, config.LogLevel)
+	pcap.SetLogLevel(log, config.LogLevel)
 
-	api, err := pcap.NewAPI(config.Buffer, *config.Agents, config.ID, config.ConcurrentCaptures)
+	api, err := pcap.NewAPI(config.Buffer, config.AgentsMTLS, config.ID, config.ConcurrentCaptures)
 	if err != nil {
-		log.Fatal("unable to create api", zap.Error(err))
+		log.Panic("Unable to create api", zap.Error(err))
 	}
 
-	api.RegisterResolver(&pcap.BoshHandler{Config: config.ManualEndpoints})
+	// set up a BoshResolver for each bosh environment
+	registerBoshResolvers(config.BoshResolverConfigs, log, api)
+
+	//TODO: CFAgentResolver
+
+	if len(api.RegisteredResolverNames(false)) == 0 {
+		log.Panic("Could not register any AgentResolvers")
+	}
 
 	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", config.Listen.Port))
 	if err != nil {
-		log.Fatal("unable to create listener", zap.Error(err))
+		log.Panic("unable to create listener", zap.Error(err))
 	}
 
-	tlsCredentials, err := cmd.LoadTLSCredentials(config.CommonConfig)
+	tlsCredentials, err := config.TLSCredentials()
 	if err != nil {
-		log.Fatal("unable to load provided TLS credentials", zap.Error(err))
+		log.Panic("unable to load provided TLS credentials", zap.Error(err))
 	}
 	server := grpc.NewServer(grpc.Creds(tlsCredentials))
 	pcap.RegisterAPIServer(server, api)
 
-	go cmd.WaitForSignal(log, api, server)
+	go pcap.WaitForSignal(log, api, server)
 
 	log.Info("starting server")
 	err = server.Serve(lis)
 	if err != nil {
-		log.Fatal("serve returned unsuccessfully", zap.Error(err))
+		log.Panic("serve returned unsuccessfully", zap.Error(err))
 	}
 
 	log.Info("serve returned successfully")
 }
+
+// registerBoshResolvers tries to register all BoshResolvers defined in configs and registers them in api.
+//
+// Logs an error if the resolver could not be initialized.
+func registerBoshResolvers(configs []pcap.BoshResolverConfig, log *zap.Logger, api *pcap.API) {
+	for _, env := range configs {
+		resolver, err := pcap.NewBoshResolver(env)
+		if err != nil {
+			log.Panic("Failed to setup BoshResolver", zap.String(pcap.LogKeyResolver, env.EnvironmentAlias), zap.Error(err))
```

Comment:
The remaining question is: is our assumption that anything in the config _must_ connect and the API will not start otherwise? If so, I'd happily bubble up the error and say there that init failed because we couldn't connect to BOSH env xzy.

Reply:
Right now this is the assumption that we make. If we decide that this is not the case you would have to add logic to check if at least one resolver works which adds more complexity. Let's keep it simple for now and fail if something in the config is unavailable.